From e78de01d2a22147f6f20dc362684ca16f55bb52e Mon Sep 17 00:00:00 2001
From: Bastien Nocera <hadess@hadess.net>
Date: Sat, 11 Jan 2025 18:56:51 +0100
Subject: [PATCH 1/2] FIX: Fix focus stealing from app when filament changes

At every filament related event, the old code would try to set the
keyboard focus to the "Device" tab, leading to a number of, either,
focus stealing events, or notifications if the desktop has focus
stealing prevention (like GNOME).

Disable the focus setting on Linux to prevent that focus stealing.

Closes: https://github.com/bambulab/BambuStudio/issues/3046
---
 src/slic3r/GUI/MainFrame.cpp | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/slic3r/GUI/MainFrame.cpp b/src/slic3r/GUI/MainFrame.cpp
index b8a80420b85f..bf23b33fea39 100644
--- a/src/slic3r/GUI/MainFrame.cpp
+++ b/src/slic3r/GUI/MainFrame.cpp
@@ -1110,10 +1110,10 @@ void MainFrame::init_tabpanel()
             m_topbar->DisableUndoRedoItems();
         }
 #endif
-
+#ifndef __WXGTK__
         if (panel)
             panel->SetFocus();
-
+#endif
         /*switch (sel) {
         case TabPosition::tpHome:
             show_option(false);
-- 
2.48.1


From 0f94d8de8189ade837abed38c4c3843b9d361391 Mon Sep 17 00:00:00 2001
From: Bastien Nocera <hadess@hadess.net>
Date: Tue, 4 Feb 2025 17:20:46 +0100
Subject: [PATCH 2/2] FIX: Fix focus stealing from printer disconnection

Every time one of the connected printers disconnects, the
the old code would try to set the keyboard focus to the "Device" tab,
leading to a number of, either, focus stealing events, or notifications
if the desktop has focus stealing prevention (like GNOME).

Disable the focus setting on Linux to prevent that focus stealing.

See https://github.com/bambulab/BambuStudio/pull/5891#issuecomment-2635143655
---
 src/slic3r/GUI/StatusPanel.cpp | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/src/slic3r/GUI/StatusPanel.cpp b/src/slic3r/GUI/StatusPanel.cpp
index 484b101205d4..0848978c9a74 100644
--- a/src/slic3r/GUI/StatusPanel.cpp
+++ b/src/slic3r/GUI/StatusPanel.cpp
@@ -4255,7 +4255,9 @@ void StatusPanel::set_default()
     m_ams_control_box->Hide();
     m_ams_control->Reset();
     error_info_reset();
+#ifndef __WXGTK__
     SetFocus();
+#endif
 }
 
 void StatusPanel::show_status(int status)
-- 
2.48.1

